src/main/java/com/agnieszkapawska/flashcards/domain/facades/LearningFacade.java

```diff
@@ -0,0 +1,122 @@
+        shouldReturnFromMethod = false;
+        Flashcard flashcard = flashcardService.findById(flashcardId);
+
+        flashcardsToLearnService.findByFlashcardId(flashcardId)
+                .ifPresent(flashcardsToLearn -> {
+                    markAnswer(answer, flashcard, flashcardsToLearn, flashcardsToLearnService, flashcardsToRepeatService);
+                    flashcardService.saveFlashcard(flashcard);
+                });
+        if (shouldReturnFromMethod) {
+            return;
+        }
+
+        flashcardsToRepeatService.findByFlashcardIs(flashcardId)
+                .ifPresent(flashcardsToRepeat -> {
+                    markAnswer(answer, flashcard, flashcardsToRepeat, flashcardsToRepeatService, flashcardsToRefreshService);
+                    flashcardService.saveFlashcard(flashcard);
+                });
+        if (shouldReturnFromMethod) {
+            return;
+        }
+
+        flashcardsToRefreshService.findByFlashcardId(flashcardId)
+                .ifPresent(flashcardsToRefresh -> {
+                    markAnswer(answer, flashcard, flashcardsToRefresh, flashcardsToRefreshService, flashcardsToRefreshService);
+                    flashcardService.saveFlashcard(flashcard);
+                });
+    }
+
+    private void markAnswer(
+            Answer answer,
+            Flashcard flashcard,
+            FlashcardsStorage flashcardsStorage,
+            FlashcardsStorageService currentFlashcardsStorageService,
+            FlashcardsStorageService futureFlashcardsStorageService) {
+
+        if(answer.getIsCorrect()) {
+            if (flashcard.getCorrectAnswerCounter() < 2) {
+                flashcard.setCorrectAnswerCounter(flashcard.getCorrectAnswerCounter() + 1);
+            } else {
+                if(flashcardsStorage instanceof FlashcardsToRefresh) {
+                    shouldReturnFromMethod = true;
+                    return;
+                }
+                moveFlashcardFromOneFlashcardsStorageToAnother(
+                        answer, flashcard, flashcardsStorage, currentFlashcardsStorageService, futureFlashcardsStorageService);
+            }
+        } else {
+            flashcard.setCorrectAnswerCounter(0);
+            if (flashcardsStorage instanceof FlashcardsToLearn) {
+                shouldReturnFromMethod = true;
+                return;
+            }
+            moveFlashcardFromOneFlashcardsStorageToAnother(
+                    answer, flashcard, flashcardsStorage, currentFlashcardsStorageService, flashcardsToLearnService);
+        }
+        shouldReturnFromMethod = true;
+    }
+
+    private void moveFlashcardFromOneFlashcardsStorageToAnother(
+            Answer answer, Flashcard flashcard, FlashcardsStorage fromWhichFlashcardsStorageIsMoved,
+            FlashcardsStorageService currentFlashcardsStorageService, FlashcardsStorageService futureFlashcardsStorageService) {
+        Long userId = Long.parseLong(answer.getUserId());
```

Comment:
Passing bigger object to method only to get from is one fields it's not good idea. Let's already pass userId as parameter.